### PR TITLE
Makes `_Nothing` a singleton

### LINKED ISSUE #693: Make `_Nothing.__new__` a real singleton
Currenlty, it is still possible to produce another `Nothing` instance.
Example: 

```python
NewNothing = _Nothing()
```

https://github.com/tek/amino/blob/master/amino/maybe.py#L163-L168

### PR DESCRIPTION
**Pull request by thepabloaguilar**
# Makes `_Nothing` a singleton

## Checklist

- [X] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [X] I have created at least one test case for the changes I have made
- [X] I have added my changes to the `CHANGELOG.md`

## Related issues

Closes #693

### CODE REVIEW

File: returns/maybe.py [hunk 1]
```diff
@@ -285,6 +285,12 @@ class _Nothing(Maybe[Any]):
     """Represents an empty state."""
 
     _inner_value: None
+    _instance: Optional['_Nothing'] = None
+
+    def __new__(cls, *args: Any, **kwargs: Any) -> '_Nothing':
+        if _Nothing._instance is None:  # noqa: WPS437
```
Comment (sobolevn) [inline returns/maybe.py:291]: Shouldn't it be `cls._instance`?

**Comment by sobolevn** [comment]
Thanks a lot! 👍

File: returns/maybe.py (hunk 1 above)
Comment (thepabloaguilar) [reply returns/maybe.py:291]: Yes, it's better!!

File: returns/maybe.py [hunk 2]
```diff
@@ -285,6 +285,12 @@ class _Nothing(Maybe[Any]):
     """Represents an empty state."""
 
     _inner_value: None
+    _instance: Optional['_Nothing'] = None
+
+    def __new__(cls, *args: Any, **kwargs: Any) -> '_Nothing':
+        if cls._instance is None:  # noqa: WPS437
```
Comment (sobolevn) [inline returns/maybe.py:291]: Do we still need `noqa` here?

Comment (thepabloaguilar) [reply returns/maybe.py]: Nop

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks!

**Pull request merged**